frontend/stmt_inferrer.py

```diff
@@ -0,0 +1,296 @@
+"""Inferrer for python statements.
+
+Infers the types for the following expressions:
+    - Assign(expr* targets, expr value)
+    - AugAssign(expr target, operator op, expr value)
+    - Return(expr? value)
+    - Delete(expr* targets)
+    - If(expr test, stmt* body, stmt* orelse)
+    - While(expr test, stmt* body, stmt* orelse)
+    - For(expr target, expr iter, stmt* body, stmt* orelse)
+    - AsyncFor(expr target, expr iter, stmt* body, stmt* orelse)
+    - With(withitem* items, stmt* body)
+    - AsyncWith(withitem* items, stmt* body)
+    - Try(stmt* body, excepthandler* handlers, stmt* orelse, stmt* finalbody)
+
+    TODO:
+    - Import(alias* names)
+    - ImportFrom(identifier? module, alias* names, int? level)
+    - Global(identifier* names)
+    - Nonlocal(identifier* names)
+"""
+
+import expr_inferrer as expr, ast, sys
+from context import Context
+from i_types import *
+
+def __infer_assignment_target(target, context, value_type):
```

Comment:
Right, but the distinction between private and protected applies only for methods of a class, right? And here we're not inside a class, so this could at most mean that some function is not supposed to be used by code outside this module. Then I'm pretty sure one underscore is enough.